src/components/tabs.jsx

```diff
@@ -1,82 +1,141 @@
-import React, { useState, Fragment } from 'react';
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+  useRef,
+  Children,
+} from 'react';
 import PropTypes from 'prop-types';
+import { v1 } from 'uuid';
+
 import '../styles/components/tabs.scss';
 
-const Title = ({ children, active, onClick }) => {
-  const handleKeyPress = (key) => {
-    if (key.key === 'Enter') {
-      onClick();
+// This is just a configuration component, it doesn't need to render anything as
+// it will be used by a <Tabs> component
+export const Tab = () => null;
+Tab.propTypes = {
+  /**
+   * Title of that tab
+   */
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  /**
+   * Optional ID for that tab, one of the expected options for the parent <Tabs> component
+   */
+  id: PropTypes.string,
+  /**
+   * Content of that tab
+   */
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
+    .isRequired,
+  /**
+   * Choose that tab as the default to be displayed
+   */
+  defaultSelected: PropTypes.bool,
+};
+
+const tabType = PropTypes.shape({
+  type: PropTypes.oneOf([Tab]),
+});
+
+export const Tabs = ({ children, active }) => {
+  const idRef = useRef(v1());
+
+  const isManaged = typeof active !== 'undefined';
+  const tabs = Children.toArray(children).map((child, index) => ({
+    title: child.props.title,
+    id: typeof child.props.id === 'undefined' ? `${index}` : child.props.id,
+    children: child.props.children,
+    defaultSelected: child.props.defaultSelected,
+  }));
+
+  const [selectedState, setSelectedState] = useState(() => {
+    if (isManaged) {
+      return active;
     }
-  };
-  return (
-    <div
-      data-testid="tab-title"
-      role="button"
-      tabIndex={0}
-      onClick={onClick}
-      onKeyPress={key => handleKeyPress(key)}
-      className={`tabs__header__item ${active ? 'tabs__header__item--active' : ''}`}
-    >
-      {children}
-    </div>
+    const defaultSelected = tabs.find(tab => tab.defaultSelected);
```

Comment:
As it's not impossible to accidentally specify `defaultSelected` on more than one tabs, I wonder if it might be worth either throwing an error or logging it in the console if it's the case.

Reply:
A warning has now been added